### PR TITLE
Module to check privileges in JS

### PR DESCRIPTION
**Pull request by SmittieC**
[Ticket](https://dimagi-dev.atlassian.net/browse/SC-2610)

## Technical Summary

Added a `privileges` module that we can use to check privileges in JS scripts, similar to how we check toggles like [here](https://github.com/dimagi/commcare-hq/blob/336a78662d4992edb4ccd796d7c464b07925ab3a/corehq/apps/data_interfaces/static/data_interfaces/js/case_property_input.js#L55). Going forward, this will hopefully make it easier to check domain privileges in JS. Currently, knowlege about a domain having a _specific_ privilege is passed to JS using a specific flag for that privilege. Each privilege requires a separate flag being passed from the view to JS. This is not ideal.

The change in this PR automatically passes in all privileges to JS, so we don't have to worry about the JS side having knowlegde about a specific privilege. We can just use the privilege module in JS to figure out what privileges there are.
## Safety Assurance
Impact on page load times is not certain. For the Dimagi Enterprise plan, there are about 53 privileges being passed to the FE, so 53 additional fields.

### Safety story

Local testing

### QA Plan

No QA plan

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### Example usage in JS
Screenshot from 2023-05-11 09-40-35

### CODE REVIEW

File: corehq/util/context_processors.py [hunk 1]
```diff
@@ -138,6 +138,25 @@ def js_toggles(request):
...
+        domain = request.project.name
+    elif getattr(request, 'domain', None):
+        domain = request.domain
+
+    if not domain:
+        return {}
+
+    plan_version = Subscription.get_subscribed_plan_by_domain(domain)
+    return {
+        'privileges': list(get_privileges(plan_version)),
```
Comment (SmittieC) [inline corehq/util/context_processors.py:156]: If we're worried about performance impact due to passing all privileges down, we can add a filter here to only pass those down that we care about in the JS code. When checking a new privilege in JS, we would then just need to update this filter as well. Other suggestions welcome.

**Review by kaapstorm — COMMENTED** [review]
This looks good to me. I'm curious to hear how others would have implemented it.

File: corehq/util/context_processors.py (hunk 1 above)
Comment (orangejenny) [reply corehq/util/context_processors.py:153]: Always good to think about, but I'm not worried about this - looks like there are ~50 privileges, and we're already passing down ~200 feature flags. In general we pass an enormous amount of js around, this probably isn't the most valuable place to optimize.

**Comment by orangejenny** [comment]
looks good to me

**Review by kaapstorm — APPROVED** [review]

**Review by Charl1996 — APPROVED** [review]

**Pull request merged**